somaticseq/combine_callers.py

```diff
@@ -0,0 +1,632 @@
+        
+        snv_intermediates.append(snv_lofreq_out)
+        indel_intermediates.append(indel_lofreq_out)
+        intermediate_vcfs['LoFreq']['snv']   = snv_varscan_out
+        intermediate_vcfs['LoFreq']['indel'] = indel_varscan_out
+
+    if scalpel:
+        
+        if exclusion:
+            scalpel_ex = bed_exclude(scalpel, exclusion, os.sep.join(( outdir, 'scalpel.ex.vcf')) )
+            intermediate_files.add(scalpel_ex)
+        else:
+            scalpel_ex = scalpel
+        
+        if inclusion:
+            scalpel_in = bed_include(scalpel_ex, inclusion, os.sep.join(( outdir, 'scalpel.in.vcf')) )
+            intermediate_files.add(scalpel_in)
+        else:
+            scalpel_in = scalpel_ex
+
+        scalpel_out = os.sep.join(( outdir, 'indel.scalpel.vcf' ))
+        copy_TextFile.copy(scalpel_in, scalpel_out)
+        intermediate_files.add(scalpel_out)
+        indel_intermediates.append(scalpel_out)
+        
+    if strelka:
+        import vcfModifier.modify_ssStrelka as mod_strelka
+        
+        if exclusion:
+            strelka_ex = bed_exclude(strelka, exclusion, os.sep.join(( outdir, 'strelka.ex.vcf')) )
+            intermediate_files.add(strelka_ex)
+        else:
+            strelka_ex = strelka
+        
+        if inclusion:
+            strelka_in = bed_include(strelka_ex, inclusion, os.sep.join(( outdir, 'strelka.in.vcf')) )
+            intermediate_files.add(strelka_in)
+        else:
+            strelka_in = strelka_ex
+
+        snv_strelka_out   = os.sep.join(( outdir, 'snv.strelka.vcf' ))
+        indel_strelka_out = os.sep.join(( outdir, 'indel.strelka.vcf' ))
+        mod_strelka.convert(strelka_in, snv_strelka_out, indel_strelka_out)
+        
+        for file_i in snv_strelka_out, indel_strelka_out:
+            intermediate_files.add( file_i )
+        
+        snv_intermediates.append(snv_strelka_out)
+        indel_intermediates.append(indel_strelka_out)
+        intermediate_vcfs['Strelka']['snv']   = snv_strelka_out
+        intermediate_vcfs['Strelka']['indel'] = indel_strelka_out
+
+
+    # Combine SNV/INDEL variant candidates
+    snv_combined   = os.sep.join(( outdir, 'unsorted.CombineVariants.snv.vcf' ))
+    indel_combined = os.sep.join(( outdir, 'unsorted.CombineVariants.indel.vcf' ))
+    
+    getUniqueVcfPositions.combine(snv_intermediates, snv_combined)
+    getUniqueVcfPositions.combine(indel_intermediates, indel_combined)
+    for file_i in snv_combined, indel_combined:
+        intermediate_files.add( file_i )
+    
+    # Sort them:
+    snv_combined_sorted = os.sep.join(( outdir, 'CombineVariants.snv.vcf' ))
+    indel_combined_sorted = os.sep.join(( outdir, 'CombineVariants.indel.vcf' ))
+    
+    vcfsorter(ref, snv_combined,   snv_combined_sorted)
+    vcfsorter(ref, indel_combined, indel_combined_sorted)
+    
+
+    if not keep_intermediates:
+        for file_i in intermediate_files:
+            subprocess.call( ('rm', '-v', file_i ) )
```

Comment:
Using something like `logger` in combination with `os.remove` would be better option. When failures happen, throwing exceptions would be more informative in a Python call instead of a `subprocess.call`.